sklearn/metrics/classification.py

```diff
@@ -2038,3 +2038,55 @@ def brier_score_loss(y_true, y_prob, sample_weight=None, pos_label=None):
     y_true = np.array(y_true == pos_label, int)
     y_true = _check_binary_probabilistic_predictions(y_true, y_prob)
     return np.average((y_true - y_prob) ** 2, weights=sample_weight)
+
+
+def calibration_loss(y_true, y_prob, bin_size=2):
+    """Compute Calibration score by bins.
+    The calibration loss is defined as the measure to access the quality of
+    learning methods and learned models. A calibration measure based on
+    overlaping binning is CAL (Caruana and Niculescu-Mizil, 2004).
+
+    Parameters
+    ----------
+    y_true : array, shape (n_samples,)
+        True targets.
+     y_prob : array, shape (n_samples,)
+        Probabilities of the positive class.
+     bin_size : int
+        Size of the bin (samples) analysed in one iteration
+     Returns
+    -------
+    score : float
+        Calibration loss
+     Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import calibration_loss
+    >>> y_true = np.array([0, 1, 1, 0])
+    >>> y_true_categorical = np.array(["spam", "ham", "ham", "spam"])
+    >>> y_prob = np.array([0.1, 0.9, 0.8, 0.3])
+    >>> calibration_loss(y_true, y_prob, bin_size = 1)
+    0.175
+    >>> calibration_loss(y_true, y_prob, bin_size = 3)
+    0.83333333333333348
+    """
+    pos_loss = 0.0
+    neg_loss = 0.0
+    for bin_start in range(0, len(y_true) - bin_size + 1):
+        bin_end = bin_start + bin_size
+        actual_per_pos_class = (y_true[bin_start:bin_end]
+                                .sum()) / bin_size
+        bin_error_pos = abs(y_prob[bin_start:bin_end]
+                            - actual_per_pos_class).sum()
+        pos_loss += bin_error_pos
+        actual_per_neg_class = (bin_size - y_true[bin_start:bin_end]
+                                .sum()) / bin_size
+        bin_error_neg = abs((1-y_prob[bin_start:bin_end])
+                            - actual_per_neg_class).sum()
+        neg_loss += bin_error_neg
+
+    pos_loss /= (len(y_true) - bin_size + 1)
+    neg_loss /= (len(y_true) - bin_size + 1)
+    loss = (0.5) * (pos_loss + neg_loss)
+
```

Comment:
IMO it would better save the len(y_true) in a variable to don't call len() some times

Reply:
Done